packages/circuit-ui/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright 2025, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use client';
+
+import {
+  flip,
+  offset as offsetMiddleware,
+  type Placement,
+  size,
+  type SizeOptions,
+  useFloating,
+} from '@floating-ui/react-dom';
+import {
+  type ComponentType,
+  forwardRef,
+  Fragment,
+  type KeyboardEvent,
+  useCallback,
+  useEffect,
+  useId,
+  useRef,
+  useState,
+} from 'react';
+
+import { Dialog, type DialogProps } from '../Dialog/Dialog.js';
+import type { ClickEvent } from '../../types/events.js';
+import { useStackContext } from '../StackContext/index.js';
+import { useMedia } from '../../hooks/useMedia/index.js';
+import { sharedClasses } from '../../styles/shared.js';
+import { applyMultipleRefs } from '../../util/refs.js';
+import { clsx } from '../../styles/clsx.js';
+
+import classes from './Popover.module.css';
+
+export interface PopoverReferenceProps {
+  'onClick': (event: ClickEvent) => void;
+  'onKeyDown'?: (event: KeyboardEvent) => void;
+  'id': string;
+  'aria-haspopup': boolean;
+  'aria-controls': string;
+  'aria-expanded': boolean;
+}
+
+export interface PopoverProps
+  extends Omit<
+    DialogProps,
+    | 'open'
+    | 'onCloseEnd'
+    | 'onCloseStart'
+    | 'isModal'
+    | 'animationDuration'
+    | 'preventClose'
+    | 'initialFocusRef'
+    | 'preventOutsideClickRefs'
+  > {
+  /**
+   * The initial state of the Popover.
+   */
+  initialOpen?: boolean;
```

Comment:
Let's use `defaultOpen` to match the Toggletip component (inspired by the [`HTMLInputElement`'s `defaultValue` attribute](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/defaultValue)).
```suggestion
  defaultOpen?: boolean;
```